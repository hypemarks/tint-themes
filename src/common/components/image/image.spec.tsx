import { render } from '@testing-library/react';
import React from 'react';
import Image from './image';

describe('Test <Image /> Component', () => {
  it('should match snapshot', () => {
    const wrapper = render(<Image src={'fake-src'} displayType={'normal'} alt={'image'} />);
    expect(wrapper.asFragment()).toMatchSnapshot();
  });

  it('should contain passed className', () => {
    const className = 'className';
    const wrapper = render(<Image className={className} displayType={'normal'} src={'fake-src'} alt={'image'} />);
    const image = wrapper.queryByTestId('image');
    expect(image).toBeInTheDocument();
    expect(image).toHaveClass(className);
  });
});
