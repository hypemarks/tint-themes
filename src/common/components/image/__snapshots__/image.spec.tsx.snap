// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Image /> Component should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="image  image--normal image--loading"
    data-testid="image"
  >
    <img
      alt="image"
      data-src="fake-src"
    />
  </div>
</DocumentFragment>
`;
